### PR TITLE
Single chain caprieval

### LINKED ISSUE #989: Add global backbone RMSD to caprieval
## **Desired feature/enhancement**

Add a global RMSD metric calculated over the entire backbone of all chains.

## **Motivation**

Currently `caprieval` only calculate RMSDs if there are more than one chain present.
It would be nice to add a global RMSD metric, i.e. calculate over the entire backbone of all chains. In that way also when using haddock in single molecule mode (e.g. for sampling conformations) we could still compare the models to a reference.

## **Description**

Add a new column to the output of caprieval with the global RMSD.
Alternatively, the ligand RMSD could be calculated over the entire backbone (or a defined region) when only one molecule is present.

### PR DESCRIPTION
**Pull request by VGPReys**
You are about to submit a new Pull Request. Before continuing make sure you read the [contributing guidelines](CONTRIBUTING.md) and that you comply with the following criteria:

- [ ] You have sticked to Python. Please talk to us before adding other programming languages to HADDOCK3
- [ ] Your PR is about CNS
- [ ] Your code is well documented: proper docstrings and explanatory comments for those tricky parts
- [ ] You structured the code into small functions as much as possible. You can use classes if there is a (state) purpose
- [ ] Your code follows our coding style
- [ ] You wrote tests for the new code
- [ ] `tox` tests pass. *Run `tox` command inside the repository folder*
- [ ] `-test.cfg` examples execute without errors. *Inside `examples/` run `python run_tests.py -b`*
- [ ] PR does not add any dependencies, unless permission granted by the HADDOCK team
- [ ] PR does not break licensing
- [ ] Your PR is about writing documentation for already existing code :fire:
- [ ] Your PR is about writing tests for already existing code :godmode:

---

Closes #989 by adding a new parameter `global_rmsd` in `[caprieval]`, where the full structure RMSD is now computed, regardless of any reference/ligands chains.
Still a sequence alignment is performed.

Additionally, few modifications were made on the uni/integration tests for code optimization and readability.

### CODE REVIEW

File: src/haddock/modules/analysis/caprieval/defaults.yaml [hunk 1]
```diff
@@ -62,6 +62,19 @@ dockq:
   group: analysis
   explevel: easy
 
+global_rmsd:
+  default: true
+  type: boolean
+  title: Calculate global RMSD
+  short: Calculate full structure RMSD.
```
Comment (amjjbonvin) [inline src/haddock/modules/analysis/caprieval/defaults.yaml:69]: Does this mean all heavy atoms RMSD?

Comment (VGPReys) [reply src/haddock/modules/analysis/caprieval/defaults.yaml:69]: No, this means backbone RMSD (with default parameters), with the exception of small-molecules
For all heavy atoms RMSD, one should turn on the `allatoms = true` parameter.
Maybe I should update the long description to emphasize this behavior

**Comment by amjjbonvin** [comment]
Probably a good idea to update the long description indeed

**Review by amjjbonvin — DISMISSED** [review]

File: src/haddock/modules/analysis/caprieval/capri.py [hunk 2]
```diff
@@ -507,6 +508,42 @@ def calc_fnat(self, cutoff: float = 5.0) -> None:
                 self.fnat = len(intersection) / float(len(ref_contacts))
         else:
             log.warning("No reference contacts found")
+    
+    def calc_global_rmsd(self) -> None:
```
Comment (rvhonorato) [inline src/haddock/modules/analysis/caprieval/capri.py:512]:
```suggestion
    # TODO: Move to (future) `librmsd.py`
    def calc_global_rmsd(self) -> None:
```

**Review by rvhonorato — CHANGES_REQUESTED** [review]
Do not change the tests, specially the helper functions, without a good justification. Please revert the changes to the integration tests's helper functions since they are unrelated to the PR.

The actual feature looks good to me 👍🏽

**Comment by VGPReys** [comment]
> Do not change the tests, specially the helper functions, without a good justification. Please revert the changes to the integration tests's helper functions since they are unrelated to the PR.

Good reasons are:
- it took me forever to understand the code as variable names did not make sense.
- added some comments for next user.
- allowed myself for slight optimization on the way.

**Comment by rvhonorato** [comment]
Sure, but this has nothing to do with the current feature. It should have been another PR.

 I won't be too nitpicky about it but at least add a description to the PR - and keep this in mind for the next time. :)

File: src/haddock/modules/analysis/caprieval/capri.py [hunk 3]
```diff
@@ -713,34 +755,36 @@ def add_chain_from_segid(pdb_path: PDBPath) -> Path:
...
             file_name = jm.name if isinstance(jm, Path) else jm.file_name
             if m == file_name:
                 # add the data
-                j.irmsd = capri_dic[m]["irmsd"]
-                j.fnat = capri_dic[m]["fnat"]
-                j.lrmsd = capri_dic[m]["lrmsd"]
-                j.ilrmsd = capri_dic[m]["ilrmsd"]
-                j.dockq = capri_dic[m]["dockq"]
+                for target_key in target_keys:
+                    j.__setattr__(target_key, capri_dic[m][target_key])
```
[the remark this answers is not in the document]
Comment (VGPReys) [reply src/haddock/modules/analysis/caprieval/capri.py:787]: yes ok, that one may be a little advanced for some people

Comment (VGPReys) [inline src/haddock/modules/analysis/caprieval/capri.py:787]:
```suggestion
                   # Set a new target_key attribute to object with capri_dic[m][target_key] value
                    j.__setattr__(target_key, capri_dic[m][target_key])
```

**Review by rvhonorato — APPROVED** [review]

**Review by amjjbonvin — APPROVED** [review]

**Pull request merged**